File: frontend-hq-portal/src/types/department.ts
export interface Department {
  departmentId: number;
  accountId: number;
  departmentName: string;
  departmentCode?: string;
  enabled: boolean;
}

export interface DepartmentSummary {
  departmentId: number;
  accountId: number;
  departmentCode?: string | null;
  departmentName: string;
  description?: string | null;
  revenueCenterCode?: string | null;
  isSubDepartment?: boolean | null;
  parentDepartmentId?: number | null;
  enabled: boolean;
  modifiedDate: string;
  modifiedBy: string;
}

export interface UpsertDepartmentPayload {
  departmentCode?: string | null;
  departmentName: string;
  description?: string | null;
  revenueCenterCode?: string | null;
  isSubDepartment?: boolean | null;
  parentDepartmentId?: number | null;
}

File: backend/EWHQ.Api/Models/Entities/Department.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace EWHQ.Api.Models.Entities;

[Table("Department")]
public class Department
{
    [Key]
    [Column(Order = 0)]
    public int DepartmentId { get; set; }

    [Key]
    [Column(Order = 1)]
    public int AccountId { get; set; }

    [MaxLength(100)]
    [Required]
    public string DepartmentName { get; set; } = string.Empty;

    public bool Enabled { get; set; }

    public DateTime CreatedDate { get; set; }

    [MaxLength(50)]
    [Required]
    public string CreatedBy { get; set; } = string.Empty;

    public DateTime ModifiedDate { get; set; }

    [MaxLength(50)]
    [Required]
    public string ModifiedBy { get; set; } = string.Empty;

    [MaxLength(200)]
    public string? Description { get; set; }

    [MaxLength(50)]
    public string? DepartmentCode { get; set; }

    [MaxLength(50)]
    public string? RevenueCenterCode { get; set; }

    public bool? IsSubDepartment { get; set; }

    public int? ParentDepartmentId { get; set; }

}
